File: model/figure.go
package model

type Contour []Point

func NewFigure(contour Contour, position Point, color RGB) Figure {
	return &figure{
		contour:  contour,
		position: position,
		color:    color,
	}
}

type Figure interface {
	Contour() Contour
	Position() Point
	Color() RGB
	SetPosition(position Point)
}

type figure struct {
	contour  Contour
	position Point
	color    RGB
}

func (f *figure) Contour() Contour {
	return f.contour
}

func (f *figure) Position() Point {
	return f.position
}

func (f *figure) Color() RGB {
	return f.color
}

func (f *figure) SetPosition(position Point) {
	f.position = position
}

File: web/2/src/Figure.ts
// Point.ts
export interface Point {
    x: number;
    y: number;
}

// RGB.ts
export interface RGB {
    r: number;
    g: number;
    b: number;
}

// Contour.ts
export type Contour = Point[];

// Figure.ts
export interface Figure {
    getContour(): Contour;
    getPosition(): Point;
    getColor(): RGB;
    setPosition(position: Point): void;
}

export class FigureImpl implements Figure {
    private contour: Contour;
    private position: Point;
    private color: RGB;

    constructor(contour: Contour, position: Point, color: RGB) {
        this.contour = contour;
        this.position = position;
        this.color = color;
    }

    getContour(): Contour {
        return this.contour;
    }

    getPosition(): Point {
        return this.position;
    }

    getColor(): RGB {
        return this.color;
    }

    setPosition(position: Point): void {
        this.position = position;
    }
}
